windows-packaging/CalicoWindows/kubernetes/kube-proxy-service.ps1

```diff
@@ -43,8 +43,17 @@ if ($kubeProxyVer -match "v([0-9])\.([0-9]+)") {
 }
 
 # Determine the windows version and build number for DSR support.
-$OSInfo = (Get-ComputerInfo  | select WindowsVersion, OsBuildNumber)
-$PlatformSupportDSR = (($OSInfo.WindowsVersion -as [int]) -GE 1903 -And ($OSInfo.OsBuildNumber -as [int]) -GE 18317)
+# OsHardwareAbstractionLayer is a version string like 10.0.17763.1432
+$OSInfo = (Get-ComputerInfo  | select WindowsVersion, OsBuildNumber, OsHardwareAbstractionLayer)
+
+# Windows supports DSR if
+# - it is 1809 build 1432
+# - it is 1903 or later
+$min1809BuildSupportingDSR = (($OSInfo.OsHardwareAbstractionLayer.Split(".") | select-object -Last 1) -as [int]) -GE 1432
+$windows1809 = (($OSInfo.WindowsVersion -as [int]) -EQ 1809 -And ($OSInfo.OsBuildNumber -as [int]) -GE 17763)
+$windows1903OrNewer = (($OSInfo.WindowsVersion -as [int]) -GE 1903 -And ($OSInfo.OsBuildNumber -as [int]) -GE 18317)

```

Comment:
Do you think Wiindows 2004 or 2022 has a greater build number than 18317?

Reply:
Yes, Windows 2004 is `19041`, 20H2 is `19042`, and Windows 2022 is `20348`: https://docs.microsoft.com/en-us/windows/release-health/windows-server-release-info